### PR TITLE
Support https parameters

### PR DESCRIPTION
**Pull request by ganmacs**
Signed-off-by: Yuta Iwama <ganmacs@gmail.com>

### CODE REVIEW

File: lib/fluent/plugin/out_prometheus_pushgateway.rb [hunk 1]
```diff
@@ -47,6 +70,21 @@ def configure(conf)
         super
 
         @push_client = ::Prometheus::Client::Push.new("#{@job_name}:#{fluentd_worker_id}", @instance, @gateway)
+
+        use_tls = gateway && (URI.parse(gateway).scheme == 'https')
+
+        if use_tls
+          # prometheus client doesn't have an interface to set the HTTPS options
+          http = @push_client.instance_variable_get(:@http)
```
Comment (cosmo0920) [inline lib/fluent/plugin/out_prometheus_pushgateway.rb]: OMG!
Should we report this monkey patch as feature request to prometheus ruby client?

Comment (ganmacs) [reply lib/fluent/plugin/out_prometheus_pushgateway.rb]: I already did. https://github.com/prometheus/client_ruby/pull/181

File: lib/fluent/plugin/out_prometheus_pushgateway.rb [hunk 2]
```diff
@@ -17,6 +17,14 @@
 require 'prometheus/client/push'
 require 'fluent/plugin/output'
 
+begin
+  require 'fluent/tls'
+rescue LoadError
+  # compatible layer for fluentd v1.9.1 or earlier
+  # https://github.com/fluent/fluentd/pull/2802
+  require 'fluent/plugin/tls'
```
Comment (repeatedly) [inline lib/fluent/plugin/out_prometheus_pushgateway.rb]: should be `require_relative`?

Comment (ganmacs) [reply lib/fluent/plugin/out_prometheus_pushgateway.rb]: Why do you think so? An absolute path is fine for me.

Comment (repeatedly) [reply lib/fluent/plugin/out_prometheus_pushgateway.rb]: What happens if other plugin have `fluent/plugin/tls` ?

Comment (ganmacs) [reply lib/fluent/plugin/out_prometheus_pushgateway.rb]: I created own directory for prometheus_pushgateway then.

**Pull request merged**